src/ucp/tag/eager_snd.c

```diff
@@ -92,7 +92,7 @@ static size_t ucp_tag_pack_eager_sync_first_dt(void *dest, void *arg)
     hdr->super.total_len       = req->send.length;
     hdr->req.ep_id             = ucp_send_request_get_ep_remote_id(req);
     hdr->super.msg_id          = req->send.msg_proto.message_id;
-    hdr->req.reqptr            = (uintptr_t)req;
+    hdr->req.req_id            = ucp_send_request_get_id( req);
```

Comment:
- extra space
- maybe whenever we pack both ep_id and req_id we can check the flag on ep_id directly ? or even add something like ucp_ep_pack_request_hdr() to pack both ep_id and req_id ?

Reply:
- ep_id/req_id are initialized differently in different protocols, so we can't pass EP and request as parameters in common function, so IMHO better to initialize them directly than define specific initializer for each protocol.
 - direct/indirect req_id has to be defined by direct/indirect **local** ep_id, local and remote EP IDs can be initialized differently if we add config parameter